sdk/packages/eternum/src/scripts/extract.ts

```
@@ -0,0 +1,97 @@
+    const content = fs.readFileSync(filePath, "utf8");
+
+    // Pattern to match full JSDoc blocks with examples
+    const methodPattern = /\/\*\*\s*([\s\S]*?)\*\/\s*public\s+async\s+([a-zA-Z_]+)/g;
+    const paramPattern = /@param\s+props\.([a-zA-Z_]+)\s*-\s*([^\n]+)/g;
+    const examplePattern = /@example[\s\S]*?```typescript\n([\s\S]*?)```/;
+
+    const examples: Example[] = [];
+    let match: RegExpExecArray | null;
+
+    // Extract all examples
+    while ((match = methodPattern.exec(content)) !== null) {
```

Biome (1.9.4)

[error] The assignment should not be in an expression.

The use of assignments in expressions is confusing.
Expressions are often considered as side-effect free.

(lint/suspicious/noAssignInExpressions)